element_calcium_imaging/imaging_withpostprocessing.py

```diff
@@ -0,0 +1,1206 @@
+    """
+
+    # Run processing only on Scan with ScanInfo inserted
+    @property
+    def key_source(self):
+        return ProcessingTask & scan.ScanInfo
+
+    def make(self, key):
+        task_mode, output_dir = (ProcessingTask & key).fetch1(
+            "task_mode", "processing_output_dir"
+        )
+
+        output_dir = find_full_path(get_imaging_root_data_dir(), output_dir).as_posix()
+        if not output_dir:
+            output_dir = ProcessingTask.infer_output_dir(key, relative=True, mkdir=True)
+            # update processing_output_dir
+            ProcessingTask.update1(
+                {**key, "processing_output_dir": output_dir.as_posix()}
+            )
+
+        if task_mode == "load":
+            method, imaging_dataset = get_loader_result(key, ProcessingTask)
+            if method == "suite2p":
+                if (scan.ScanInfo & key).fetch1("nrois") > 0:
+                    raise NotImplementedError(
+                        f"Suite2p ingestion error - Unable to handle"
+                        f" ScanImage multi-ROI scanning mode yet"
+                    )
+                suite2p_dataset = imaging_dataset
+                key = {**key, "processing_time": suite2p_dataset.creation_time}
+            elif method == "caiman":
+                caiman_dataset = imaging_dataset
+                key = {**key, "processing_time": caiman_dataset.creation_time}
+            else:
+                raise NotImplementedError("Unknown method: {}".format(method))
+        elif task_mode == "trigger":
+
+            method = (ProcessingParamSet & key).fetch1("processing_method")
+
+            image_files = (scan.ScanInfo.ScanFile & key).fetch("file_path")
+            image_files = [
+                find_full_path(get_imaging_root_data_dir(), image_file)
+                for image_file in image_files
+            ]
+
+            if method == "suite2p":
+                import suite2p
+
+                suite2p_params = (ProcessingParamSet & key).fetch1("params")
+                suite2p_params["save_path0"] = output_dir
+                suite2p_params["fs"] = (scan.ScanInfo & key).fetch1("fps")
+
+                input_format = pathlib.Path(image_files[0]).suffix
+                suite2p_params["input_format"] = input_format[1:]
+
+                suite2p_paths = {
+                    "data_path": [image_files[0].parent.as_posix()],
+                    "tiff_list": [f.as_posix() for f in image_files],
+                }
+
+                suite2p.run_s2p(ops=suite2p_params, db=suite2p_paths)  # Run suite2p
+
+                _, imaging_dataset = get_loader_result(key, ProcessingTask)
+                suite2p_dataset = imaging_dataset
+                key = {**key, "processing_time": suite2p_dataset.creation_time}
+
+            elif method == "caiman":
+                from element_interface.run_caiman import run_caiman
+
+                params = (ProcessingParamSet & key).fetch1("params")
```

Comment:
Above, a similar statement is used for `suite2p_params` - could this be moved outside the `method` if/else case bock? If there's a reason to prefer keeping here, I would differentiate this line as `caiman_params`

Reply:
Nice catch! But for the consistency of the following lines in the suite2p part (as well as original suite2p jargon) they are better fetched separately. `params` is `caiman_params` now.